kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -362,6 +366,28 @@ protected void close() {
                 currentConnectedGroup.remove(clientHost);
             }
 
+            // Try to remove all stored groupID on the metadata store.
+            if (log.isDebugEnabled()) {
+                log.debug("Try to remove all stored groupID on the metadata store. Current connected clientIds: {}",
+                        currentConnectedClientId);
+            }
+            currentConnectedClientId.forEach(clientId -> {
+                String path = groupIdStoredPath + GroupIdUtils.groupIdPathFormat(clientHost, clientId);
+                metadataStore.delete(path, Optional.empty())
```

Comment:
so when we close a client connection we do the deletion.
I am +1 to this as a work around.

We should this store data somewhere else, like we are doing now with the ProducerId

I don't have time these days, but I will be happy to work on a alternative solution in the mid term